lib/compress.js

```diff
@@ -5414,6 +5414,7 @@ merge(Compressor.prototype, {
         // p(){return x;} ---> p:()=>x
         if (compressor.option("arrows")
             && compressor.parent() instanceof AST_Object
+            && !self.value.uses_arguments
```

Comment:
How about `.uses_eval` to guard against `eval("arguments[0]")`? :thinking:

Reply:
Not sure if it's needed, but added in https://github.com/mishoo/UglifyJS2/pull/2632/commits/681f771ce1f47e2e7e2bad5a31dbbedcaaa0fe6b.

Reply:
I thought `arguments` is never mangled.

Reply:
`.uses_arguments` would not be able to detect `eval("arguments[0]")`, even though `arguments` is used in this case.

Reply:
Oh - I see. Thanks.